### PR TITLE
test: construct e2e deployments through buildTestDeployment

### PR DESCRIPTION
**Pull request by ingvagabund**
Test code refactoring

### CODE REVIEW

**Comment by ingvagabund** [comment]
@fanhaouu PTAL

**Review by a7i — APPROVED** [review]
/approve

File: test/e2e/e2e_topologyspreadconstraint_test.go [hunk 1]
```diff
@@ -188,20 +188,19 @@ func TestTopologySpreadConstraint(t *testing.T) {
...
-				t.Fatalf("Error creating Deployment %s: %v", violatorDeploymentName, err)
+			if _, err := clientSet.AppsV1().Deployments(violatorDeployment.Namespace).Create(ctx, violatorDeployment, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("Error creating Deployment %s: %v", violatorDeployment.Name, err)
 			}
 			defer func() {
 				clientSet.AppsV1().Deployments(violatorDeployment.Namespace).Delete(ctx, violatorDeployment.Name, metav1.DeleteOptions{})
 				waitForPodsToDisappear(ctx, t, clientSet, violatorDeployment.Labels, violatorDeployment.Namespace)
 			}()
-			waitForPodsRunning(ctx, t, clientSet, violatorDeployment.Labels, int(violatorCount), violatorDeployment.Namespace)
+			waitForPodsRunning(ctx, t, clientSet, violatorDeployment.Labels, int(*violatorDeployment.Spec.Replicas), violatorDeployment.Namespace)
```
Comment (fanhaouu) [inline test/e2e/e2e_topologyspreadconstraint_test.go:203]: It seems like you're trying to ensure consistency in style.

**Comment by fanhaouu** [comment]
LGTM

**Comment by fanhaouu** [comment]
Apologies for the delayed response, I was on holiday for China's National Day.

**Pull request merged**